### PR TITLE
[pigeon] allow gen of unused classes

### PR DESCRIPTION
**Pull request by tarrinneal**
changes pigeon_lib to allow generation of classes that aren't referenced in any API.

### CODE REVIEW

**Comment by yaakovschectman** [comment]
Will all classes now be converted, or do they need to be marked in some way?

**Comment by tarrinneal** [comment]
> Will all classes now be converted, or do they need to be marked in some way?

All classes in the pigeon definition file will be generated.

File: packages/pigeon/pigeons/core_tests.dart [hunk 1]
```diff
@@ -18,6 +18,13 @@ enum AnotherEnum {
...
 }
 
+// This exists to show that unused data classes still generate.
+class UnusedClass {
+  UnusedClass({this.aField});
+
+  Object? aField;
+}
+
 class SimpleClass {
```
Comment (stuartmorgan) [inline packages/pigeon/pigeons/core_tests.dart:28]: Based on the output diffs, it looks like this class wasn't being generated before. That suggests that it should be unused, and thus removed.

**Review by stuartmorgan — COMMENTED** [review]
Per offline discussion, it would be good to add native unit tests that just instantiate the unused class, so that we have confirmation that it's being generated for each language.

File: packages/pigeon/pigeons/core_tests.dart (hunk 1 above)
Comment (reidbaker) [reply packages/pigeon/pigeons/core_tests.dart:28]: Or it should be used to validate that unused classes are generated? https://github.com/flutter/packages/pull/7529#pullrequestreview-2267160848

**Review by reidbaker — APPROVED** [review]

File: packages/pigeon/platform_tests/alternate_language_test_plugin/android/src/main/java/com/example/alternate_language_test_plugin/AlternateLanguageTestPlugin.java [hunk 2]
```diff
@@ -688,6 +688,10 @@ public void error(Throwable error) {
         };
     flutterSmallApiOne.echoString(aString, resultCallbackOne);
   }
+
+  public CoreTests.UnusedClass testIfUnusedClassIsGenerated() {
```
Comment (stuartmorgan) [inline packages/pigeon/platform_tests/alternate_language_test_plugin/android/src/main/java/com/example/alternate_language_test_plugin/AlternateLanguageTestPlugin.java:692]: Looks like the linter wants a `@NonNull` on this.

**Review by stuartmorgan — APPROVED** [review]
LGTM, thanks!

**Pull request merged**